app/assets/javascripts/services/miq_service.js

```diff
@@ -158,4 +158,20 @@ ManageIQ.angular.app.service('miqService', ['$timeout', '$document', '$q', 'API'
       miqService.sparkleOff();
     }
   };
+
+  this.redirectBack = function(message, flashType, redirectUrl, reload) {
+    var flash = { message: message };
+
+    flash.level = flashType;
+
+    miqFlashLater(flash);
+
+    $timeout(function() {
+      $window.location.href = redirectUrl;
+
+      if(reload) {
```

Comment:
A redirectBack alone won't do the tree update and the Tenant tree needs to be updated after ever Add/Save